assembl/static2/js/app/components/administration/saveButton.jsx

```diff
@@ -40,57 +43,101 @@ const createVariablesForMutation = (thematic) => {
   };
 };
 
-const SaveButton = ({ createThematic, deleteThematic, enabled, refetchThematics, thematics, updateThematic }) => {
+const SaveButton = ({
+  i18n,
+  createThematic,
+  deleteThematic,
+  enabled,
+  refetchThematics,
+  thematics,
+  updateThematic,
+  updateDiscussionPreference,
+  preferences,
+  languagePreferenceHasChanged,
+  resetLanguagePreferenceChanged }) => {
   const saveAction = () => {
     displayAlert('success', `${I18n.t('loading.wait')}...`);
     const promisesArray = [];
-    thematics.forEach((thematic) => {
-      if (thematic.isNew && !thematic.toDelete) {
-        // create thematic
-        const payload = {
-          variables: createVariablesForMutation(thematic)
-        };
-        const p1 = () => {
-          return createThematic(payload);
-        };
-        promisesArray.push(p1);
-      } else if (thematic.toDelete && !thematic.isNew) {
-        // delete thematic
-        const payload = {
-          variables: {
-            thematicId: thematic.id
-          }
-        };
-        const p3 = () => {
-          return deleteThematic(payload);
-        };
-        promisesArray.push(p3);
-      } else {
-        // update thematic
-        const variables = createVariablesForMutation(thematic);
-        variables.id = thematic.id;
-        const payload = {
-          variables: variables
-        };
-        const p2 = () => {
-          return updateThematic(payload);
-        };
-        promisesArray.push(p2);
-      }
-    });
+    // Compare the redux state change to the saved data in apollo
+
+    if (languagePreferenceHasChanged) {
+      // Save and update the apolloStore
+      const payload = {
+        variables: {
+          languages: preferences
+        },
+        update: (storeProxy, { data: { updateDiscussionPreference: { preferences: { languages } } } }) => {
+          // Update the apollo cache
+          const query = storeProxy.readQuery({
+            query: getDiscussionPreferenceLanguage,
+            variables: { inLocale: i18n.locale }
+          });
+          const newData = { ...query };
+          newData.discussionPreferences.languages = languages;
+          storeProxy.writeQuery({
+            query: getDiscussionPreferenceLanguage,
+            variables: { inLocale: i18n.locale },
+            data: newData
+          });
+        }
+      };
+      updateDiscussionPreference(payload);
+      resetLanguagePreferenceChanged();
+    }
 
-    runSerial(promisesArray)
-      .then(() => {
-        refetchThematics();
-        displayAlert('success', I18n.t('administration.successThemeCreation'));
-      })
-      .catch((error) => {
-        displayAlert('danger', `${error}`, false, 30000);
+    if (enabled) {
```

Comment:
Can you rename `enabled` by `thematicsHasChanged`

Reply:
Done